lib/api/CAnomalyJobConfig.cc

```diff
@@ -677,6 +682,18 @@ void CAnomalyJobConfig::CAnalysisConfig::parse(const rapidjson::Value& analysisC
     m_CategorizationFieldName = parameters[CATEGORIZATION_FIELD_NAME].fallback(EMPTY_STRING);
     m_CategorizationFilters = parameters[CATEGORIZATION_FILTERS].fallback(TStrVec{});
 
+    const std::string& modelPruneWindowString{
+        parameters[MODEL_PRUNE_WINDOW].fallback(EMPTY_STRING)};
+
+    if (modelPruneWindowString.empty() == false) {
+        m_ModelPruneWindow = CAnomalyJobConfig::CAnalysisConfig::durationSeconds(
+            modelPruneWindowString, core_t::TTime{0});
+
+        // Ensure that the model prune window is never smaller than the bucket span.
+        const core_t::TTime minModelPruneWindow{m_BucketSpan};
+        m_ModelPruneWindow = std::max(m_ModelPruneWindow, minModelPruneWindow);
```

Comment:
It's probably best to log a warning here instead of silently changing the user's choice.